### PR TITLE
Windows-compatible programmatic compilations

### PR DESCRIPTION
**Pull request by tansongyang**
So far, this contains just the changes needed to successfully compile on Windows. See commit comments for details on why I switched from `sane` to `chokidar`.

I tried to fix `ts:precompile`, but I wasn't entirely sure what it was supposed to do. Running `ember ts:precompile` on `master` right now produces this error:

```
Error: Command failed: tsc --outDir Me\AppData\Local\Temp/e-c-ts-precompile-23496 --rootDir Me\ember-cli-typescript --allowJs false --noEmit false --declaration --sourceMap false --inlineSourceMap false --inlineSources false

tests/dummy/app/helpers/typed-help.ts(7,1): error TS4082: Default export of the module has or is using private name 'Ember.Helper'.
```

So, I'll need some help fixing `ts:precompile`.

TLDR:

- [ ] Fix `ts:precompile`.
- [ ] Clean up. In particular, we should decide what we want `reportDiagnostic` and `reportWatchStatus` to do.

### CODE REVIEW

**Comment by chriskrycho** [comment]
Heh, that specific failure looks like a problem with the type definitions, not with your work! (More on Monday.)

**Comment by mike-north** [comment]
I ran into some mega annoying install issues due to chokidar's need for native dependencies, and 

This library was explicitly avoided in ember-cli, due to reasons stated in the following issue
https://github.com/ember-cli/ember-cli/issues/4503

Also react-native avoided adding it for similar reasons
https://github.com/facebook/react-native/issues/628

TL;DR - we may want to work our way back to using watchman
